### ISSUE TITLE
Timed out tests doesn't fail the sbt task

### ISSUE THREAD

**Issue by durban** [typelevel/cats-effect#3503]
This might be a specs2 thing, but it seems unlikely that it's intentional.

- One example is [here](https://github.com/typelevel/cats-effect/issues/3463#issuecomment-1448605353)
- Another one [here](https://github.com/typelevel/cats-effect/actions/runs/4443535679/jobs/7800886809) (the CI execution was cancelled, not failed; I don't know what and why cancelled it, might it be related?):
```
[info] DrainBatchSpec
[info] Batch draining should
[info]   o work correctly in the presence of concurrent stealers
[info] timeout exception
```

**Comment by armanbilge** [comment]
> * CI execution was cancelled, not failed; I don't know what and why cancelled

We have a 1 hour time limit on CI to prevent hanging. Maybe it's not sufficient?

https://github.com/typelevel/cats-effect/blob/fb35b3f85d03cb5d33eb5e12ceb4d3f179346783/.github/workflows/ci.yml#L106

**Comment by durban** [comment]
Hm, okay, that's what cancelled it :-) There might be a real problem there (most JVM runs finished in around 15 minutes, so 60 should be plenty), unclear if related to the timed out specs2 test...

**Comment by durban** [comment]
It _might_ be intentional, but I still don't understand _why_ they make a timeout a skipped test:
https://github.com/etorreborre/specs2/blob/SPECS2-4.19.2/core/shared/src/main/scala/org/specs2/specification/core/Execution.scala#L156-L160

Also, there is this here:
https://github.com/typelevel/cats-effect/blob/series/3.x/tests/shared/src/test/scala/cats/effect/Runners.scala#L103-L117
So it might be an unintentional interaction...

**Comment by armanbilge** [comment]
> It _might_ be intentional, but I still don't understand _why_ they make a timeout a skipped test:

Also, there is this commit: https://github.com/etorreborre/specs2/commit/24766396e5643900b76a039c15c218e5bd0d8fec 🤔

**Comment by durban** [comment]
Should be fixed by  #3551.
